stock_reserve/models/stock_reserve.py

```diff
@@ -0,0 +1,218 @@
+    want to override:
+
+    * company_id
+    * location_id
+    * location_dest_id
+
+    Optionally, you may be interested to define:
+
+    * date_validity  (once passed, the reservation will be released)
+    * note
+    """
+
+    _name = "stock.reservation"
+    _description = "Stock Reservation"
+    _inherits = {"stock.move": "move_id"}
+
+    move_id = fields.Many2one(
+        "stock.move",
+        "Reservation Move",
+        required=True,
+        readonly=True,
+        ondelete="cascade",
+        index=True,
+    )
+    date_validity = fields.Date("Validity Date")
+
+    @api.model
+    def default_get(self, fields_list):
+        """Fix default values
+
+        - Ensure default value of computed field `product_qty` is not set
+          as it would raise an error
+        - Compute default `location_id` based on default `picking_type_id`.
+          Note: `default_picking_type_id` may be present in context,
+          so code that looks for default `location_id` is implemented here,
+          because it relies on already calculated default
+          `picking_type_id`.
+        """
+        # if there is 'location_id' field requested, ensure that
+        # picking_type_id is also requested, because it is required
+        # to compute location_id
+        if "location_id" in fields_list and "picking_type_id" not in fields_list:
+            fields_list = fields_list + ["picking_type_id"]
+
+        res = super().default_get(fields_list)
+
+        if "product_qty" in res:
+            del res["product_qty"]
+
+        # At this point picking_type_id and location_id
+        # should be computed in default way:
+        #     1. look up context
+        #     2. look up ir_values
+        #     3. look up property fields
+        #     4. look up field.default
+        #     5. delegate to parent model
+        #
+        # If picking_type_id is present and location_id is not, try to find
+        # default value for location_id
+        if not res.get("picking_type_id", None):
+            res["picking_type_id"] = self._default_picking_type_id()
+
+        picking_type_id = res.get("picking_type_id")
+        if picking_type_id and not res.get("location_id", False):
+            picking = self.env["stock.picking"].new(
+                {"picking_type_id": picking_type_id}
+            )
+            picking.onchange_picking_type()
+            res["location_id"] = picking.location_id.id
+        if "location_dest_id" in fields_list:
+            res["location_dest_id"] = self._default_location_dest_id()
+        if "product_uom_qty" in fields_list:
+            res["product_uom_qty"] = 1.0
+        return res
+
+    @api.model
+    def get_location_from_ref(self, ref):
+        """Get a location from a xmlid if allowed
+        :param ref: tuple (module, xmlid)
+        """
+        try:
+            location = self.env.ref(ref, raise_if_not_found=True)
+            location.check_access_rule("read")
```

Comment:
I am not sure this is enough. In multicompany, you can pass access rule check but fail record rule I think. At least in multicompany test this is what I'm getting.